cpu/sam0_common/periph/pwm.c

```diff
@@ -17,80 +17,42 @@
  *
  * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
  *
  * @}
  */
 
-#include <stdint.h>
-#include <string.h>
-
-#include "log.h"
 #include "cpu.h"
 #include "board.h"
 #include "periph/gpio.h"
 #include "periph/pwm.h"
 
+/* determine the number of CC channels */
+#if defined(TCC_SYNCBUSY_CC5)
+#define CC_NUMOF    6
+#elif defined(TCC_SYNCBUSY_CC4)
+#define CC_NUMOF    5
+#elif defined(TCC_SYNCBUSY_CC3)
+#define CC_NUMOF    4
+#elif defined(TCC_SYNCBUSY_CC2)
+#define CC_NUMOF    3
+#elif defined(TCC_SYNCBUSY_CC1)
+#define CC_NUMOF    2
+#elif defined(TCC_SYNCBUSY_CC0)
+#define CC_NUMOF    1
+#endif
+
```

Comment:
Can't we use TCCx_CC_NUM to determine the number of CC instead ?

Reply:
You mean instead of `CC_NUMOF`?

Reply:
Ah I see what you mean

Reply:
Thank you, that uncovered a bug - I miss-understood CCB and the amout of Capture/Compare channels is not uniform across timers.